build.cake

```diff
@@ -37,39 +37,48 @@ Task("Clean")
 // RESTORE PACKAGES
 //////////////////////////////////////////////////////////////////////
 
-Task("InitializeBuild")
+Task("NuGet-Package-Restore")
     .Does(() =>
     {
-      if (BuildSystem.IsRunningOnAppVeyor)
-      {
-          var tag = AppVeyor.Environment.Repository.Tag;
-
-          if (tag.IsTag)
-          {
-              packageVersion = tag.Name;
-          }
-          else
-          {
-              var buildNumber = AppVeyor.Environment.Build.Number;
-              packageVersion = version + "-CI-" + buildNumber + dbgSuffix;
-              if (AppVeyor.Environment.PullRequest.IsPullRequest)
-                  packageVersion += "-PR-" + AppVeyor.Environment.PullRequest.Number;
-              else if (AppVeyor.Environment.Repository.Branch.StartsWith("release", StringComparison.OrdinalIgnoreCase))
-                  packageVersion += "-PRE-" + buildNumber;
-              else
-                  packageVersion += "-" + AppVeyor.Environment.Repository.Branch;
-          }
-
-          AppVeyor.UpdateBuildVersion(packageVersion);
-      }
+        NuGetRestore(SOLUTION);
     });
 
 //////////////////////////////////////////////////////////////////////
+// INITIALIZE BUILD
+//////////////////////////////////////////////////////////////////////
+
+Setup(context =>
```

Comment:
I'm not a Cake expert, but it seems like this is something you'd want to explicitly run no matter what target you choose or how you set the different target dependencies up, for which `Setup()` fits the bill quite nicely.